IdentityCore/src/telemetry/request_telemetry/MSIDLastRequestTelemetrySerializedItem.m

```diff
@@ -0,0 +1,110 @@
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#import "MSIDCurrentRequestTelemetrySerializedItem+Internal.h"
+#import "MSIDLastRequestTelemetrySerializedItem.h"
+
+@interface MSIDLastRequestTelemetrySerializedItem()
+
+@property (nonatomic) NSArray<MSIDRequestTelemetryErrorInfo *> *errorsInfo;
+
+@end
+
+@implementation MSIDLastRequestTelemetrySerializedItem
+
+- (instancetype)initWithSchemaVersion:(NSNumber *)schemaVersion defaultFields:(NSArray *)defaultFields errorInfo:(NSArray *)errorsInfo platformFields:(NSArray *)platformFields
+{
+    self = [super initWithSchemaVersion:schemaVersion defaultFields:defaultFields platformFields:platformFields];
+    if (self)
+    {
+        _errorsInfo = errorsInfo;
+    }
+    return self;
+}
+
+// Builds last telemetry string using default serialization of each set of fields
+// specified in the last telemetry string schema
+- (NSString *)serialize
+{
+    NSString *telemetryString = [NSString stringWithFormat:@"%@|", self.schemaVersion];
+    telemetryString = [telemetryString stringByAppendingFormat:@"%@|", [super serializeFields: super.defaultFields]];
+    
+    NSUInteger startLength = [telemetryString lengthOfBytesUsingEncoding:NSUTF8StringEncoding];
+    telemetryString = [telemetryString stringByAppendingFormat:@"%@|", [self serializeErrorsInfoWithCurrentStringSize:startLength]];
+    
+    telemetryString = [telemetryString stringByAppendingFormat:@"%@", [super serializeFields: self.platformFields]];
+    
+    return telemetryString;
+}
+
+#pragma mark Helper
+
+- (NSString *)serializeErrorsInfoWithCurrentStringSize:(NSUInteger)startLength
+{
+    NSString *failedRequestsString = @"";
+    NSString *errorMessagesString = @"";
+    
+    if (self.errorsInfo.count > 0)
+    {
+        int lastIndex = (int)self.errorsInfo.count - 1;
+        
+        // Set first post in fencepost structure -- last item in string doesn't have comma at the end
+        NSString *currentFailedRequest = [NSString stringWithFormat:@"%ld,%@", self.errorsInfo[lastIndex].apiId, self.errorsInfo[lastIndex].correlationId ?: @""];
+        NSString *currentErrorMessage = [NSString stringWithFormat:@"%@", self.errorsInfo[lastIndex].error ?: @""];
+        
+        // Only add error info into string if the resulting string smaller than 4KB
+        if ([currentFailedRequest lengthOfBytesUsingEncoding:NSUTF8StringEncoding] +
```

Comment:
nit: helper function might be helpful here since this code is a bit duplicated.

Reply:
Acknowledged. I'll return to this point after this release, in subsequent telemetry work. Added to my to-do list.